Objects/dictobject.c

```diff
@@ -3322,22 +3323,37 @@ dictiter_new(PyDictObject *dict, PyTypeObject *itertype)
 {
     dictiterobject *di;
     di = PyObject_GC_New(dictiterobject, itertype);
-    if (di == NULL)
+    if (di == NULL) {
         return NULL;
+    }
     Py_INCREF(dict);
     di->di_dict = dict;
     di->di_used = dict->ma_used;
-    di->di_pos = 0;
     di->len = dict->ma_used;
-    if (itertype == &PyDictIterItem_Type) {
+    if (itertype == &PyDictRevIterKey_Type ||
+        itertype == &PyDictRevIterItem_Type ||
+        itertype == &PyDictRevIterValue_Type) {
+        if (dict->ma_used) {
+            di->di_pos = dict->ma_keys->dk_nentries - 1;
+        }
+        else {
+            di->di_pos = 0;
+        }
+    }
+    else {
+        di->di_pos = 0;
+    }
```

Comment:
This if statements looks ugly.

```
di->di_pos = 0;

if ((itertype == &PyDictRevIterKey_Type ||
     itertype == &PyDictRevIterItem_Type ||
     itertype == & PyDictRevIterValue_Type) && di->ma_used > 0)
{
    di->di_pos = dict->ma_keys->dk_nentries - 1;
}
```

This looks better, IMO.

This is matter of taste.  If you prefer current code, no need to rewrite.

Reply:
This is indeed far better, thanks for noticing that